src/walle-lib/Environment.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <vector>
+#include <iostream>
+#include "Plantation.hpp"
+#include "Obstacle.hpp"
+
+class Environment
+{
+protected:
+    double m_width{10};
+    double m_height{10};
+    // liste de zones à planter obstacles et autres grace aux coordonnées
+    // Pour les plantations il faudrait savoir l'état de la plantation pour savoir quel robot utiliser
+    std::vector<Obstacle> m_list_obstacles;
```

Comment:
1) std::vector is not a list, the m_list_obstacles is thus misleading.
A vector stores elements in a continuous memory with constant access time by index. 
A list stores elements in a discontinuous structures (eg. a linked list or more complicated) , addition is done in constant time, but access by index is has a linear complexity (it is needed to traverse the list to read "index" i). 

More generally, modern code editor, have type deduction so there is no need to duplicate the 'type' of the data in its name. I would generally prefer using the convention consisting to use "s" to indicated there is multiple element.
```cpp
std::vector<Element> elements;
// or std::list<Element> elements;
// or std::map<std::string, Element> elements;
```